### PR TITLE
[OBMIN-432] Create Favourite advertisement entity and DB structure

### PR DESCRIPTION
**Pull request by demus-n**
- created table **favorite_advertisement**
- added new field **favoriteAdvertisements** to User
- added test

### CODE REVIEW

File: src/main/resources/liquibase/db.changelog-1.2.xml [hunk 1]
```diff
@@ -68,4 +68,36 @@
             <column name="is_validated_email"/>
         </dropColumn>
     </changeSet>
+
+    <changeSet id="1.2.5: Create a table 'favorite_advertisement'" author="lavniuzhenkova">
+        <createTable tableName="favorite_advertisement">
+            <column name="id" type="${type.uuid}" valueComputed="${uuid_function}">
```
Comment (IllyaZuew) [inline src/main/resources/liquibase/db.changelog-1.2.xml:74]: maybe you should create combine primary key(user_id and advertisement_id) insted of creating independed id

Comment (demus-n) [reply src/main/resources/liquibase/db.changelog-1.2.xml]: that was a good idea, I did it

**Review by IllyaZuew — APPROVED** [review]

File: src/main/java/space/obminyashka/items_exchange/model/User.java [hunk 2]
```diff
@@ -81,6 +81,12 @@ public class User extends BaseEntity implements UserDetails {
             inverseJoinColumns = @JoinColumn(name = "blocked_id", referencedColumnName = "id"))
     private List<User> blacklistedUsers;
 
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(name = "favorite_advertisement",
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/model/User.java:85]: Why don't you use plural here "favorite_advertisements"?

File: src/main/resources/liquibase/db.changelog-1.2.xml [hunk 3]
```diff
@@ -68,4 +68,37 @@
             <column name="is_validated_email"/>
         </dropColumn>
     </changeSet>
+
+    <changeSet id="1.2.5: Create a table 'favorite_advertisement'" author="lavniuzhenkova">
```
Comment (rpkyrych) [inline src/main/resources/liquibase/db.changelog-1.2.xml:72]: And plural here as well

File: src/main/resources/liquibase/db.changelog-1.2.xml [hunk 4]
```diff
@@ -68,4 +68,37 @@
...
+        <createTable tableName="favorite_advertisement">
```
Comment (rpkyrych) [inline src/main/resources/liquibase/db.changelog-1.2.xml:73]: And here

File: src/test/java/space/obminyashka/items_exchange/service/FavoriteAdvertisementDBTest.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+@DBRider
+class FavoriteAdvertisementDBTest {
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
```
Comment (rpkyrych) [inline src/test/java/space/obminyashka/items_exchange/service/FavoriteAdvertisementDBTest.java:22]: With JUnit 5 you can use Lombok's RequiredAllArgsConstructor and to inject fields without Autowired

**Review by rpkyrych — DISMISSED** [review]

**Review by Sershok — APPROVED** [review]

**Review by ennnbaev — APPROVED** [review]

**Review by rpkyrych — APPROVED** [review]

**Review by IllyaZuew — APPROVED** [review]

**Review by Sershok — APPROVED** [review]

**Pull request merged**